x-pack/plugins/licensing/common/license.ts

```diff
@@ -169,10 +183,13 @@ export class License implements ILicense {
   }
 
   getFeature(name: string) {
-    if (!this.featuresMap.has(name)) {
-      this.featuresMap.set(name, new LicenseFeature(name, this.features[name], this));
+    let feature = this.featuresMap.get(name);
+
+    if (!feature) {
+      feature = new LicenseFeature(name, this.features[name], this);
```

Comment:
why we pass `license` in `LicenseFeature` if no one reads it?

Reply:
During license objectification, it is possible for the consumer to swap out the implementation, so we need the instance for that.

Reply:
could you provide an example? I don't see that `LicenseFeature` was created outside of `License` class

Reply:
Sorry, I don't mean swapping of the `LicenseFeature` implementation, I mean swapping how a `LicenseFeature` is serialized. This happens here:

https://github.com/elastic/kibana/pull/44922/files#diff-d2c03a616ca4d2ea0b8c58f8a58e2c82R11